src/homework1.py

```diff
@@ -0,0 +1,124 @@
+import sys
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.tree import DecisionTreeClassifier
+
+
+def main():
+    df = pd.read_csv(
+        "../src/iris.data",
+        names=[
+            "sepal length (cm)",
+            "sepal width (cm)",
+            "petal length (cm)",
+            "petal width (cm)",
+            "class",
+        ],
+    )
+
+    for i in df.columns[:4]:
+        print(i)
+        print("Mean: " + str(np.mean(df[i])))
+        print("Min: " + str(np.min(df[i])))
+        print("Max: " + str(np.max(df[i])))
+        print("Q1: " + str(np.quantile(df[i], 0.25)))
+        print("Q2: " + str(np.quantile(df[i], 0.50)))
+        print("Q3: " + str(np.quantile(df[i], 0.75)))
+        print()
+
+    fig = px.scatter(
+        df,
+        title="Scatter plot - Sepal Width vs Sepal Length",
+        x="sepal length (cm)",
+        y="sepal width (cm)",
+        color="class",
+    )
+    fig.show()
+
+    fig = px.violin(
+        df, title="Violin plot - Sepal Width", x="class", y="sepal width (cm)"
+    )
+    fig.show()
```

Comment:
Also, I remember during class the professor mentioned about writing the plots into HTML. Here is the example code from the slide for reference:
fig.write_html(file="name.html", include_plotlyjs="cdn")
However, when I ran your file, it automatically opens the plots in web browser in html form. This suggestion is optional.

Reply:
I think Dr. Pierret said that opening the plots in the web browser is ok, so I will stick with what I have. Thank you for the suggestion though and that is helpful to have in the future.